lib/pg_ha_migrations/relation.rb

```diff
@@ -96,9 +96,23 @@ def partitions(include_sub_partitions: false, include_self: false)
   end
 
   class Index < Relation
-    # TODO: implement shortening to ensure < 63 bytes
+    MAX_NAME_SIZE = 63 # bytes
+
     def self.from_table_and_columns(table, columns)
```

Comment:
This function is always used for constructing a new index name (i.e., as opposed to getting existing indexes)?

Reply:
Currently yes, although the class itself has a `valid?` method which can be used to check for both the existence and validity of an index. This brings up a really good point about the consistency of all of these structs. Like for tables, `from_table_name` expects the table to exist (and will raise an error if it doesn't). And the optional lock mode on a relation can be both an existing lock and a future target lock. It might be worthwhile to refactor all of this to be more consistent, although I don't have good sense of how to do that (and it most definitely should be in a different PR)